custom/ewsghana/stock_data.py

```diff
@@ -0,0 +1,51 @@
+from corehq.apps.commtrack.models import SupplyPointCase
+from corehq.apps.locations.models import SQLLocation
+from custom.ewsghana.models import EWSGhanaConfig
+from custom.logistics.stock_data import StockDataSynchronization
+from custom.logistics.tasks import sync_stock_transactions_for_facility
+from dimagi.utils.couch.database import iter_docs
+
+
+class EWSStockDataSynchronization(StockDataSynchronization):
+
+    @property
+    def apis(self):
+        return (
+            ('stock_transaction', sync_stock_transactions_for_facility),
+        )
+
+    @property
+    def all_stock_data(self):
+        return EWSGhanaConfig.for_domain(self.domain).all_stock_data
+
+    def get_location_id(self, facility):
+        sp = SupplyPointCase.view('hqcase/by_domain_external_id',
+                                  key=[self.domain, str(facility)],
+                                  reduce=False,
+                                  include_docs=True,
+                                  limit=1).first()
+        return sp.location_id
+
+    def get_ids(self):
+        supply_points_ids = SQLLocation.objects.filter(
+            domain=self.domain,
+            location_type__administrative=False
+        ).order_by('created_at').values_list('supply_point_id', flat=True)
+        return [doc['external_id'] for doc in iter_docs(SupplyPointCase.get_db(), supply_points_ids)]
```

Comment:
(not actionable) it would be nice if we had this field in SQL to avoid this call...